src/c.ml

```diff
@@ -34,17 +34,24 @@ module Kind = struct
       cxx_version_introduced ~obj ~dune_version ~version_introduced:(1, 10)
     | _ -> Unrecognized
 
-  let possible_fns t fn ~dune_version =
-    match t with
-    | C -> [fn ^ ".c"]
+  let possible_exts ~dune_version = function
+    | C -> [".c"]
     | Cxx ->
-      let cxx = [fn ^ ".cpp"] in
+      let exts = [".cpp"] in
+      let exts =
+        if dune_version >= (1, 10) then
```

Comment:
If it was worth a version bump then, is it worth a version bump now? Should it say `>= (1, 11)`?

Reply:
Version bump for what? To control the behavior of copying with line directives? I think it's a bit overkill as it's a non breaking change. Unlike introducing new extensions for source files.

Reply:
Uh, OK, I didn't notice the line directives change, but I think I noticed something else.

I claim `possible_fns` at the base of this PR *never* returns `".cc"`, even when syntax is `(1, 10)` (because it goes into the ">= (1,8)" branch and exits the function).

You're fixing that, and I'm saying if the changes in the set of files are worth protecting by the version number, we should probably change this to (1, 11).

Reply:
> I claim possible_fns at the base of this PR never returns ".cc", even when syntax is (1, 10) (because it goes into the ">= (1,8)" branch and exits the function).

Okay, I see what you mean. That is indeed another bug that I'm fixing. I will make note of it in the CHANGES.

> You're fixing that, and I'm saying if the changes in the set of files are worth protecting by the version number, we should probably change this to (1, 11).

I still think the versions should stay as intended. We already support the .cc extension whenever the dune lang >= (1, 10). The only issue is that we don't report a correct error message. So the change that you've pointed out is simply fixing an error message, and I think that we should fix the error message unconditionally.

Reply:
Oh, I see, that's what I missed. Thanks!